privacyidea/static/components/login/controllers/loginControllers.js

```diff
@@ -102,19 +102,24 @@ angular.module("privacyideaApp")
     $scope.myCountdown = "";
     // We save the previous State in the $rootScope, so that we
     // can return there
-    $rootScope.$on('$stateChangeSuccess',
-        function (ev, to, toParams, from, fromParams) {
-            //debug: console.log("we changed the state from " + from + " to " + to);
-            //debug: console.log(from);
-            //debug: console.log(fromParams);
-            //debug: console.log(to);
+    $transitions.onBefore({},
+        function () {
+            // The stateParams or $state.params are always changed as a reference.
+            // So we need to do a deep copy, to preserve it between state transitions
+            var oldParams = new Array();
+            for (let [key, value] of Object.entries($state.params)) {
+                oldParams[key] = value;
+            }
             $rootScope.previousState = {
-                state: from.name,
-                params: fromParams
+                state: $state.current.name,
+                params: oldParams
             };
-
+        });
+    $transitions.onSuccess({},
+        function() {
```

Comment:
Since the `checkReloadListeners()` function currently doesn't really do anything, maybe we can get rid of it.

Reply:
You are probably right. I would like to put this in an extra issue and commit, to make this change easier trackable.
Will be done in #2127